dask/dataframe/io/hdf.py

```diff
@@ -295,26 +300,23 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, division, lock)
             raise ValueError("Start row number ({}) is above or equal to stop "
                              "row number ({})".format(start, stop))
 
-        if division:
-            dsk = {(name, 0): (_pd_read_hdf, path, key, lock,
-                               base)}
+        def update(s):
+            new = base.copy()
+            new.update({'start': s, 'stop': s + chunksize})
+            return new
```

Comment:
See `toolz.merge`

Reply:
Ah, I see that you're just copying this code from before.  No obligation to change it, but it would be welcome if you did.